From c4e3f1fe764c211c4052611135b107a1fe6913be Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 5 Dec 2024 15:23:15 +0800
Subject: [PATCH 182/274] AOSCOS: drm: radeon: revise driver_for_each_device()
 usage in DPMS handler methods

Commit 448d105120b1 ("drm/amdgpu: Remove dead static variable") removed
the static pointer `pdriver'.

Revise the methods by first obtaining a `device_driver' struct via
`driver_find()' and then feeding it back to `driver_for_each_device()',
which expects a `device_driver' struct as its first argument.

Fixes: "AOSCOS: MIPS: Loongson: Add a hook to turn on/off DPMS when Fn+F7 is pressed"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/gpu/drm/radeon/radeon_drv.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/radeon/radeon_drv.c b/drivers/gpu/drm/radeon/radeon_drv.c
index 55608eb4d82b..2e852e35a10c 100644
--- a/drivers/gpu/drm/radeon/radeon_drv.c
+++ b/drivers/gpu/drm/radeon/radeon_drv.c
@@ -38,6 +38,7 @@
 #include <linux/pci.h>
 
 #include <drm/drm_client_setup.h>
+#include <drm/drm_crtc_helper.h>
 #include <drm/drm_drv.h>
 #include <drm/drm_file.h>
 #include <drm/drm_fourcc.h>
@@ -639,15 +640,19 @@ static int radeon_lvds_dpms_callback(struct device *dev, void *arg)
 void radeon_lvds_dpms_off(void)
 {
 	int on = 0;
+	struct device_driver *drv;
 
-	driver_for_each_device(&pdriver->driver, NULL, &on, radeon_lvds_dpms_callback);
+	drv = driver_find("radeon", &pci_bus_type);
+	driver_for_each_device(drv, NULL, &on, radeon_lvds_dpms_callback);
 }
 
 void radeon_lvds_dpms_on(void)
 {
 	int on = 1;
+	struct device_driver *drv;
 
-	driver_for_each_device(&pdriver->driver, NULL, &on, radeon_lvds_dpms_callback);
+	drv = driver_find("radeon", &pci_bus_type);
+	driver_for_each_device(drv, NULL, &on, radeon_lvds_dpms_callback);
 }
 
 static int __init radeon_module_init(void)
-- 
2.47.1

